File: done/number-of-trailing-zeros-of-n-factorial.py
# https://www.codewars.com/kata/number-of-trailing-zeros-of-n/train/python

from unittest import TestCase

from math import log


def zeros(n):
    if n < 1:
        return 0
    upper = int(log(n, 5))
    powers = range(upper, 0, -1)
    return sum(list(map(lambda x: n // (5 ** x), powers)))


test = TestCase()
test.assertEqual(zeros(6), 1)
test.assertEqual(zeros(12), 2)
test.assertEqual(zeros(19), 3)
test.assertEqual(zeros(26), 6)
test.assertEqual(zeros(30), 7)
test.assertEqual(zeros(100), 24)

File: DotNet/Kyu5/NumberOfTrailingZerosOfNFactorial.cs
// https://www.codewars.com/kata/number-of-trailing-zeros-of-n/train/csharp

using System;

namespace DotNet;

public class NumberOfTrailingZerosOfNFactorial
{
    public static int TrailingZeros(int n)
    {
        if (n <= 1) return 0;

        int sum = 0;
        for (int index = (int)Math.Log(n, 5); index > 0; index--)
        {
            sum += n / (int)Math.Pow(5, index);
        }

        return sum;
    }

    public static void Main()
    {
        Console.WriteLine($"{TrailingZeros(5)} == 1");
        Console.WriteLine($"{TrailingZeros(12)} == 2");
        Console.WriteLine($"{TrailingZeros(25)} == 6");
        Console.WriteLine($"{TrailingZeros(531)} == 131");
    }
}